### PR TITLE
Improve My Jetpack checkout flow consistency

### PR DESCRIPTION
**Pull request by jboland88**
## Proposed changes:
This PR includes several changes to the product purchase and "start for free" flows in My Jetpack

* Products now all default to installing the standalone plugin if one is available
* All products will go to their individual product page after a purchase, if there is one available
* Bug fixed on the product table views where there was no "loading" state after clicking on start for free
* Enables the "connect after checkout" flow for free purchases that need to go through the cart (Search free)
* Fixes inconsistent post-checkout redirect behavior that was happening between product card and product table interstitial screens
* Fixes bug in free search product activation where errors were returned for no plan as the user was trying to get a free plan

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* It is best to test this PR on a fresh Jetpack install with no standalone plugins installed - the local install already has copies of all plugins installed
* Navigate to My Jetpack, test purchasing different products from My Jetpack. Note that this PR expects that all purchases are started from the My Jetpack screen by clicking on one of the "Learn More" buttons on the product cards.
* Test the "start for free" options for different product (example below).
* Screenshot 2024-02-23 at 11 51 12 AM
* Try removing all purchases, deleting all plugins and disconnecting a few times to see what happens when you purchase disconnected or make purchases without plugins installed.

**Expected behaviors while testing**
*  When purchasing a product from My Jetpack (whenever you go through the checkout) - if that product has an available standalone plugin, the standalone should be installed and activated. After the purchase, you should be redirected to the plugin page in wp-admin.
* When choosing a "start for free" option, the standalone plugin should be installed/ activated and you should be redirected to the plugin page.

**Known Issues**
* If your very first "purchase" while Jetpack is disconnected is Jetpack Search Free - you will encounter an error during the checkout process. This is being addressed in this PR: https://github.com/Automattic/wp-calypso/pull/87958
* For initial disconnected purchases, some products will link to the cloud from the license assignment thank you screen (screenshot below). This happens for backup, scan and search. This is out of scope for this PR

Screenshot 2024-02-27 at 2 55 13 PM

* On the screen pictured above, some products might link to the incorrect page in wp-admin (ex: you might buy Boost and get linked to the Backups plugin page in wp-admin), this is being addressed in this PR: https://github.com/Automattic/wp-calypso/pull/87961

### CODE REVIEW

**Review by IanRamosC — COMMENTED** [review]
I tried a few different scenarios and realized it was not consistent (with what I was expecting), so I'm leaving a few notes as I'm going through the test steps to make sure the behavior is correct:
- With a new site and using this branch, I purchased Jetpack Boost. After connecting, Boost is installed, but I was redirected to this page:  /wp-admin/admin.php?page=jetpack-boost
- Purchased Search (that was already on the menu after connection), it redirected to the Search submenu, but the standalone plugin was not installed
- Purchased VideoPress, it got installed and the page was redirected to /wp-admin/admin.php?page=jetpack#/settings?term=videopress instead of the plugin page
- Purchased Social, the plugin was correctly installed, but it was redirect to /wp-admin/admin.php?page=jetpack#/settings?term=publicize instead of the plugin page
- Purchased a few more products (including VaultPress), removed the subscription of all of them. Purchased Boost again, and got redirected to the VaultPress page
- Removed all subscription and uninstall and removed all standalone plugins. Went back to My Jetpack and "purchased" the Free version of the ones available and it was redirected to the right page after purchase.

I'll review the code in a separate comment

File: projects/packages/my-jetpack/src/products/class-backup.php [hunk 1]
```diff
@@ -209,10 +209,12 @@ public static function get_post_activation_url() {
...
 	public static function get_manage_url() {
-		if ( static::is_jetpack_plugin_active() ) {
-			return Redirect::get_url( 'my-jetpack-manage-backup' );
-		} elseif ( static::is_plugin_active() ) {
+		// check standalone first
+		if ( static::is_standalone_plugin_active() ) {
 			return admin_url( 'admin.php?page=jetpack-backup' );
+			// otherwise, check for the main Jetpack plugin
+		} elseif ( static::is_jetpack_plugin_active() ) {
+			return Redirect::get_url( 'my-jetpack-manage-backup' );
```
Comment (IanRamosC) [inline projects/packages/my-jetpack/src/products/class-backup.php:224]: I think that, for the Boost issue I mentioned above, it fell under this condition (for some reason, skipping the check above) and ended up redirecting to the Backup page.

**Review by IanRamosC — COMMENTED** [review]
The code looks good, just left one minor comment about what I mentioned above. I'm not going to approve for now, since it's not clear whether the behavior above is expected or not

**Comment by jboland88** [comment]
Thanks for the thorough test and notes @IanRamosC 🙇 

> With a new site and using this branch, I purchased Jetpack Boost. After connecting, Boost is installed, but I was redirected to this page: /wp-admin/admin.php?page=jetpack-boost

That's the boost plugin page, correct? This is the expected behavior - my goal was to send the user to the related plugin page after purchase if the plugin is installed.

> Purchased Search (that was already on the menu after connection), it redirected to the Search submenu, but the standalone plugin was not installed

Hmmm 🤔 Where did you start the purchase from? Did you click on "Search" in the wp-admin menu or did you start from the My Jetpack page and click "Learn More" on the My Jetpack card? Right now, at least for the purposes of this PR, the standalone Search plugin will only get installed if you start from My Jetpack.

> Purchased VideoPress, it got installed and the page was redirected to /wp-admin/admin.php?page=jetpack#/settings?term=videopress instead of the plugin page

Interesting... do you remember if the the VideoPress plugin installed in this instance? If it was, this is not what was intended - you should have gone to the plugin page after purchase.

> Purchased Social, the plugin was correctly installed, but it was redirect to /wp-admin/admin.php?page=jetpack#/settings?term=publicize instead of the plugin page

Same as VideoPress, you should have been redirected to the plugin page 🤔 

> Purchased a few more products (including VaultPress), removed the subscription of all of them. Purchased Boost again, and got redirected to the VaultPress page

Thats.... weird 😅 I'll check into this and see if I can reproduce it

File: projects/packages/my-jetpack/src/products/class-backup.php (hunk 1 above)
Comment (jboland88) [reply projects/packages/my-jetpack/src/products/class-backup.php:224]: When you got sent to the backup page, was it from this screen? (ignoring that this screenshot is for videopress)
Screenshot 2024-02-27 at 2 55 13 PM

If so, I think it was a separate issue that I am looking at here: https://github.com/Automattic/wp-calypso/pull/87961

**Review by IanRamosC — COMMENTED** [review]
The code looks good again and most of the cases I tested are working as expected. I did notice two small things:
- A few purchases are redirecting to Jetpack Cloud (only when it's the first purchase)
- VaultPress is redirecting to My Jetpack (we can address it in a different PR)

Here's the list of cases that I tested and their results:
- New site:
    - Purchased Search (first purchase ever). Plugin installed. Redirected to https://cloud.jetpack.com/jetpack-search/{site_slug}. This same issue happened to other products (not all of them) when it was a site's first purchase.
        - Jetpack features are already
        - Clicking on “Visit Dashboard” redirects to https://wordpress.com/ 
    - ✅ Purchased Boost. Plugin installed. Redirected to `/wp-admin/admin.php?page=jetpack-boost`
    - ✅ Purchased Stats. Plugin not installed. Redirected to `/wp-admin/admin.php?page=stats#!/stats/day/230017055?page=stats`
    - ✅ Purchased VideoPress. Plugin installed. Redirected to `/wp-admin/admin.php?page=jetpack-videopress#/`
    - ✅ Purchased Social. Plugin installed. Redirected to`/wp-admin/admin.php?page=jetpack-social`
    - ✅ Purchased Creator. Plugin installed. Redirected to `/wp-admin/admin.php?page=my-jetpack`
    - ❓Purchased VaultPress. Plugin installed. Redirected to `/wp-admin/admin.php?page=my-jetpack`
- Removed all plugins and subscriptions, then:
    - ✅ Added Search Free. Briefly after clicking on “Get Search for Free”, another button was displayed saying “Install Jetpack Search”. Plugin Installed. `/wp-admin/admin.php?page=jetpack-search`
    - ✅ Added Boost Free. Immediately redirected to `/wp-admin/admin.php?page=jetpack-boost`. Plugin installed correctly as well.
    - ❓ Purchased VaultPress again. Plugin installed. Redirected to `/wp-admin/admin.php?page=my-jetpack`

**Comment by IanRamosC** [comment]
Also, merging trunk into this branch will probably fix a few failing checks (not urgent as they are not required)

**Comment by jboland88** [comment]
> A few purchases are redirecting to Jetpack Cloud (only when it's the first purchase)

Was the redirecting to cloud happening from the license post-activation screen (ex below)?
Screenshot 2024-02-27 at 2 55 13 PM
I had this happen too when clicking the "Go to Dashboard" button. This only happened when I was purchasing while disconnected and the site gets connected after purchase. In these cases, I think we need to pass the `from` parameter further through the different post-connection steps in Calypso so we can know someone came from My Jetpack.

> VaultPress is redirecting to My Jetpack (we can address it in a different PR)

I am getting this too. It was a silly mistake in the code where I forgot to `return` for the post checkout URL in the code 🤦

**Comment by IanRamosC** [comment]
>In these cases, I think we need to pass the from parameter further through the different post-connection steps in Calypso so we can know someone came from My Jetpack.

Yep, it was from the post-activation screen.

**Review by IanRamosC — APPROVED** [review]
LGTM!

**Pull request merged**